### PR TITLE
Implement basic /email endpoint

### PR DESCRIPTION
**Pull request by jujaga**
# Description

This PR implements the basic email functionality as provided by nodemailer. The /email endpoint is effectively a restful pass-through to the nodemailer library and retains most of its function signatures.

This PR is needed in order for emails to be issued.

[[SHOWCASE-241]](https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-241)

## Types of changes

New feature (non-breaking change which adds functionality)
Documentation (non-breaking change with enhancements to documentation)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-9f0fbe-tools.pathfinder.gov.bc.ca/dashboard?id=common-hosted-email-service-pr-1

File: app/src/components/checks.js [hunk 1]
```diff
@@ -14,29 +13,28 @@ const checks = {
...
-      // connection.close();
     }
 
     return result;
   },
 
+  /** Returns a list of all endpoint connectivity states */
   getStatus: () => Promise.all([
-    checks.getSmtpStatus()
+    checks.getSmtpStatus('apps.smtp.gov.bc.cas') // TODO: move this to constants file
```
Comment (loneil) [inline app/src/components/checks.js]: apps.smtp.gov.bc.cas, typo, should be .ca?

File: app/src/routes/v1/email.js [hunk 2]
```diff
@@ -1,61 +1,50 @@
...
+    if (req.query.devMode) {
+      const result = await emailComponent.sendMailEthereal(req.body);
+      res.status(201).json(result);
+    } else {
+      const result = await emailComponent.sendMailSmtp(req.body);
+      res.status(201).json(result);
     }
   } catch (error) {
-    log.error(error.message);
+    new Problem(502, {
```
Comment (loneil) [inline app/src/routes/v1/email.js]: 502?

File: app/src/components/checks.js (hunk 1 above)
Comment (jujaga) [reply app/src/components/checks.js]: Whoops forgot to revert from testing.

File: app/src/routes/v1/email.js (hunk 2 above)
Comment (jujaga) [reply app/src/routes/v1/email.js]: Subtracted 2

**Pull request merged**